pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,8 +54,13 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
 	APIVersion string `json:"api-version,omitempty"`
+	// PreferredVersions takes a string of multiple group/version pairs (separated by comma), and informs version negotiation
+	PreferredVersions string `json:"preferred-versions,omitempty"`
```

Comment:
Thanks @deads2k. I can see the value of using `[]string` instead of `string`. But users would prefer to type `--preferred-versions=g1/v1,g2/v2` rather than  `--preferred-versions=g1/v1 --preferred-versions=g2/v2`, imo. 

@lavalamp @caesarxuchao what do you think?